editor/static/js/editor.js

```diff
@@ -1859,7 +1859,61 @@ $(document).ready(function() {
                 description = 'Vector with '+v.value.length+' '+Numbas.util.pluralise(v.value.length,'component','components');
                 break;
             case 'list':
-                description = 'List of '+v.value.length+' '+Numbas.util.pluralise(v.value.length,'item','items');
+                function get_nested_layers() {
+                    const layer_lengths = [];
+
+                    function get_layer_lengths(v) {
+                        let next_layer = v.value;
+                        if (next_layer.length !== 0) {
+                            next_layer = next_layer[0]
+                            const nested_length = next_layer.type == "list"
+                                                  ? next_layer.value.length
+                                                  : 0;
+
+                            if (nested_length > 0) {

```

Comment:
It looks like you're trying to stop when you get to an empty list. I think that it's worth saying when you have an empty layer: `[ [ [] ] ]` is a 1×1×0 structure, while `[ [a] ]` is a 1×1 structure.

Reply:
Fixed with latest push - thanks